cmake/HPX_SetupHwloc.cmake

```diff
@@ -13,9 +13,93 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-find_package(Hwloc)
-if(NOT Hwloc_FOUND)
-  hpx_error(
-    "Hwloc could not be found, please specify Hwloc_ROOT to point to the correct location"
+if(NOT HPX_WITH_FETCH_HWLOC)
+  find_package(Hwloc)
+  if(NOT Hwloc_FOUND)
+    hpx_error(
+      "Hwloc could not be found, please specify Hwloc_ROOT to point to the correct location"
+    )
+  endif()
+else()
+  hpx_info(
+    "HPX_WITH_FETCH_HWLOC=${HPX_WITH_FETCH_HWLOC}, Hwloc will be fetched using CMake's FetchContent"
   )
+  if(UNIX)
+    include(FetchContent)
+    fetchcontent_declare(
+      HWLoc
+      URL https://download.open-mpi.org/release/hwloc/v2.9/hwloc-2.9.3.tar.gz
+      TLS_VERIFY true
+    )
+    if(NOT HWLoc_POPULATED)
+      fetchcontent_populate(HWLoc)
+      execute_process(
+        COMMAND
+          sh -c
+          "cd ${CMAKE_BINARY_DIR}/_deps/hwloc-src && ./configure --prefix=${CMAKE_BINARY_DIR}/_deps/hwloc-installed && make -j && make install"
+      )
+    endif()
+    set(HWLOC_ROOT "${CMAKE_BINARY_DIR}/_deps/hwloc-installed")
+    set(Hwloc_INCLUDE_DIR
+        ${HWLOC_ROOT}/include
+        CACHE INTERNAL ""
+    )
+    set(Hwloc_LIBRARY
+        ${HWLOC_ROOT}/lib/libhwloc.so
+        CACHE INTERNAL ""
+    )
+  elseif("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "Win64")
+    fetchcontent_declare(
+      HWLoc
+      URL https://download.open-mpi.org/release/hwloc/v2.9/hwloc-win64-build-2.9.3.zip
+      TLS_VERIFY true
+    )
+    if(NOT HWLoc_POPULATED)
+      fetchcontent_populate(HWLoc)
+    endif()
+    set(HWLOC_ROOT
+        "${CMAKE_BINARY_DIR}/_deps/hwloc-src"
+        CACHE INTERNAL ""
+    )
+    find_package(hwloc REQUIRED PATHS ${HWLOC_ROOT} NO_DEFAULT_PATH)
+    include_directories(${HWLOC_ROOT}/include)
+    link_directories(${HWLOC_ROOT}/lib)
+    set(Hwloc_INCLUDE_DIR
+        ${HWLOC_ROOT}/include
+        CACHE INTERNAL ""
+    )
+    set(Hwloc_LIBRARY
+        ${HWLOC_ROOT}/lib/libhwloc.dll.a
+        CACHE INTERNAL ""
+    )
+  else()
+    fetchcontent_declare(
+      HWLoc
+      URL https://download.open-mpi.org/release/hwloc/v2.9/hwloc-win64-build-2.9.3.zip
```

Comment:
1. It should be win32.
2. Same as above, we may have to keep a separate variable for HWLOC version.

Reply:
Could you make this change and see if it doesn't fail? I am not sure how it's related to `free(): invalid pointer`, but it should be win32